tests/unit/cli/test_main.py

```diff
@@ -26,6 +27,20 @@ def test_cli_some_command(self):
             result = runner.invoke(cli, ["local", "generate-event", "s3"])
             self.assertEqual(result.exit_code, 0)
 
+    def test_cli_with_no_region_arg_validate(self):
```

Comment:
```python

    def test_cli_with_non_standard_format_region(self):
        mock_cfg = Mock()
        with patch("samcli.cli.main.GlobalConfig", mock_cfg):
            runner = CliRunner()
            for command in ["validate", "deploy"]:
                result = runner.invoke(cli, [command, "--region", "--non-standard-format"])
                self.assertEqual(result.exit_code, 1)
                self.assertIn("Error: Provided region: --non-standard-format doesn't match a supported format",
                              result.output)

```

You could do a assertRaises too.

Reply:
Conversely if you wanted to test that `region` being set to a standard format went ok.

```python

    @patch("samcli.commands.validate.validate.do_cli")
    def test_cli_with_valid_region(self, mock_do_cli):
        mock_cfg = Mock()
        with patch("samcli.cli.main.GlobalConfig", mock_cfg):
            runner = CliRunner()
            result = runner.invoke(cli, ["validate", "--region", "us-west-2"])
            self.assertEqual(result.exit_code, 0)
        self.assertTrue(mock_do_cli.called)
        self.assertEqual(mock_do_cli.call_count, 1)
```